zk-token-sdk/src/sigma_proofs/mod.rs

```diff
@@ -15,15 +15,37 @@ pub mod pubkey_proof;
 pub mod zero_balance_proof;
 
 #[cfg(not(target_os = "solana"))]
-use {crate::errors::ProofVerificationError, curve25519_dalek::scalar::Scalar};
+use {
+    crate::errors::ProofVerificationError,
+    curve25519_dalek::{ristretto::CompressedRistretto, scalar::Scalar},
+};
 
+/// Deserializes an optional slice of bytes to a compressed Ristretto point.
+///
+/// This is a helper function for deserializing byte encodings of sigma proofs. It is designed to
+/// be used with `std::slice::Chunks`.
 #[cfg(not(target_os = "solana"))]
-fn canonical_scalar_from_slice(bytes: &[u8]) -> Result<Scalar, ProofVerificationError> {
-    if bytes.len() != 32 {
-        return Err(ProofVerificationError::Deserialization);
-    }
+fn ristretto_point_from_optional_slice(
+    optional_slice: Option<&[u8]>,
+) -> Result<CompressedRistretto, ProofVerificationError> {
+    let slice = optional_slice.ok_or(ProofVerificationError::Deserialization)?;
+    let point_bytes = slice[..32]
+        .try_into()
+        .map_err(|_| ProofVerificationError::Deserialization)?;
 
-    let scalar_bytes = bytes[..32]
+    Ok(CompressedRistretto::from_slice(point_bytes))
```

Comment:
```suggestion
```
(Sorry, github wouldn't let me make this as part of the previous suggestion because of the 1 line removed in the middle of the diff 😕 )